app/containers/Products/LinkedProductModal.tsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import {Modal, Container, Col, Row, Button} from 'react-bootstrap';
+import {JSONSchema6} from 'json-schema';
+import JsonSchemaForm from 'react-jsonschema-form';
+import {ProductRowItemContainer_product} from 'ProductRowItemContainer_product.graphql';
+import SearchDropdownWidget from 'components/Forms/SearchDropdownWidget';
+import ProductRowIdField from 'containers/Forms/ProductRowIdField';
+import FormArrayFieldTemplate from 'containers/Forms/FormArrayFieldTemplate';
+import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
+import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
+
+interface Props {
+  product: ProductRowItemContainer_product;
+  linkProductModalShow: boolean;
+  setLinkProductModalShow: (boolean) => void;
+  saveLinkedProducts: (...args: [any]) => void;
+  query: any;
+  linkData: any;
+}
+
+export const LinkedProductModalComponent: React.FunctionComponent<Props> = ({
+  product,
+  linkProductModalShow,
+  setLinkProductModalShow,
+  saveLinkedProducts,
+  query,
+  linkData
+}) => {
+  const CUSTOM_FIELDS = {
+    productRowId: (props) => {
+      return (
+        <ProductRowIdField
+          isLinkModal
+          query={props.formContext.query}
+          {...props}
+        />
+      );
+    }
+  };
+
+  const linkSchema: JSONSchema6 = {
+    type: 'array',
+    items: {
+      $ref: '#/definitions/product'
+    },
+    definitions: {
+      product: {
+        type: 'object',
+        properties: {
+          productRowId: {
+            type: 'integer'
+          }
+        }
+      }
+    }
+  };
+
+  const linkUISchema = {
+    'ui:add-text': '+',
+    'ui:remove-text': '-',
+    linkProduct: true,
+    items: {
+      'ui:field': 'product',
+      classNames: 'hidden-title',
+      productRowId: {
+        'ui:col-md': 12,
+        'ui:widget': 'SearchWidget',
+        'ui:field': 'productRowId'
+      }
+    }
+  };
```

Comment:
Those are not changing between renders, so you're better off declaring them outside of the component function, to avoid triggering unnecessary renders down the component tree